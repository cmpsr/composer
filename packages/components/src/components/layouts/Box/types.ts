export { BoxProps } from '@chakra-ui/react';
