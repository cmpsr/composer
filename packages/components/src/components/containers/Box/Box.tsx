import React, { ReactNode } from 'react';
import cn from 'classnames';

export const BOX_DEFAULT_TEST_ID = 'box';

type Props = {
  children?: ReactNode;
  className?: string;
  size?: 'full' | 'screen' | 'content';
  testId?: string;
};

export const Box = ({
  children,
  className,
  size,
  testId = BOX_DEFAULT_TEST_ID,
}: Props) => {
  return (
    <div
      className={cn(
        { 'w-screen h-screen': size === 'screen' },
        { 'w-full h-full': size === 'full' },
        { 'w-full h-full max-w-content mx-auto': size === 'content' },
        className,
      )}
      data-testid={testId}
    >
      {children}
    </div>
  );
};
