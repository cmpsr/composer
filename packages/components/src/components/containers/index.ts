export * from './Box';
export * from './Grid';
export * from './Section';
export * from './SectionChild';
