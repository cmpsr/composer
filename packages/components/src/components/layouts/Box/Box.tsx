import React, { FC } from 'react';
import { Box as ChakraBox } from '@chakra-ui/react';
import { BoxProps } from './types';

export const Box: FC<BoxProps> = (props) => <ChakraBox {...props} />;
