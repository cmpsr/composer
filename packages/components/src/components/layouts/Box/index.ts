export { Box, BoxProps } from '@chakra-ui/react';
