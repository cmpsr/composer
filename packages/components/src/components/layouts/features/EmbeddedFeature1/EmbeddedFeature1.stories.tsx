import React from 'react';
import { EmbeddedFeature1 } from '.';

export default {
  title: 'Composer/Layout/Feature/EmbeddedFeature1',
  component: EmbeddedFeature1,
};

export const withText = () => (
  <EmbeddedFeature1
    title="Create an Out-of-this-World website with this headline"
    subtitle="We are right now on the verge of finding out whether there is life elsewhere in the universe, and there are three ways we could find it."
    style={{}}
  />
);

withText.story = {
  name: 'With text',
};
