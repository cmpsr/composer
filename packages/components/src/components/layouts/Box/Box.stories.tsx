import React from 'react';
import { Meta } from '@storybook/react';
import { Box } from './Box';

export default {
  component: Box,
  title: 'Components/Layouts/Box',
} as Meta;

const Template = (args) => <Box {...args} />;
export const Playground = Template.bind({});
Playground.args = {
  children: "I'm a Box",
  border: '1px solid var(--chakra-colors-primary-default)',
  padding: 'spacer-3',
};
