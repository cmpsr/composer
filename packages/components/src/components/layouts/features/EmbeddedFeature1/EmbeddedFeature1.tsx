import React, { ReactElement } from "react";
import cn from "classnames";

import { Box } from "components/containers";
import { Asset, Typography, TypographyTypes } from "components/primitives";
import { getClassesFromStyle, getChildrenWithGrid } from "utils";

import { FeatureStyle, Asset as AssetType } from "types";

type Props = {
  title?: string;
  subtitle?: string;
  eyebrow?: string;
  embeddedAsset?: AssetType;
  className?: string;
  children?: ReactElement | ReactElement[];
  style: FeatureStyle;
};

export const EmbeddedFeature1 = ({
  embeddedAsset,
  className,
  title,
  subtitle,
  eyebrow,
  children,
  style,
}: Props) => {
  const colorClasses = getClassesFromStyle(style.background);
  return (
    <Box size="content" className={cn("items-center", colorClasses, className)}>
      <div className="text-left">
        {embeddedAsset && (
          <Asset
            asset={embeddedAsset}
            className="mt-2"
            imageClassName="object-cover"
          />
        )}
        {title && (
          <Typography
            tag="h2"
            type={TypographyTypes.Headline4}
            style={style.title}
            className="mt-4"
          >
            {title}
          </Typography>
        )}
        {eyebrow && (
          <Typography
            tag="h4"
            type={TypographyTypes.Headline6}
            style={style.eyebrow}
          >
            {eyebrow}
          </Typography>
        )}
        {subtitle && (
          <Typography
            tag="h6"
            type={TypographyTypes.Body}
            className="mt-3"
            style={style.subtitle}
          >
            {subtitle}
          </Typography>
        )}
        <div className="flex justify-start mt-4">
          {getChildrenWithGrid(children)}
        </div>
      </div>
    </Box>
  );
};
