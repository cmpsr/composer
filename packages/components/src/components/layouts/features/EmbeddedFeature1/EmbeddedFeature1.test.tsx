import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { EmbeddedFeature1 } from './EmbeddedFeature1';
import { BOX_DEFAULT_TEST_ID } from '../../../containers/Box';
import { IMAGE_DEFAULT_TEST_ID } from '../../../primitives/Image';
import {
  TYPOGRAPHY_DEFAULT_TEST_ID,
  TypographyTypes,
} from '../../../primitives/Typography';

describe('EmbeddedFeature1', () => {
  it('should match snapshot', () => {
    const hero1 = renderer.create(
      <EmbeddedFeature1 style={{}} title="title test" />,
    );
    const tree = hero1.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render text', () => {
    const { getByText } = render(
      <EmbeddedFeature1 style={{}} title="title test" />,
    );
    const element = getByText('title test');
    expect(element).toBeInTheDocument();
  });
  test('should render a Box component with class', () => {
    const { getByTestId } = render(
      <EmbeddedFeature1 style={{}} title="title test" className="foo" />,
    );
    const element = getByTestId(BOX_DEFAULT_TEST_ID);
    expect(element.classList).toContain('foo');
  });
  test('should render an embeddedAsset asset', () => {
    const embeddedAsset = {
      contentType: 'image',
      title: 'Test Image',
      url: '#',
    };
    const { getByTestId } = render(
      <EmbeddedFeature1
        embeddedAsset={embeddedAsset}
        style={{}}
        title="title test"
        className="foo"
      />,
    );

    const element = getByTestId(IMAGE_DEFAULT_TEST_ID) as HTMLSourceElement;
    expect(element.type).toBe('image/webp');
  });
  test('should render a Typography component when title is added', () => {
    const embeddedAsset = {
      contentType: 'image',
      title: 'Test Image',
      url: '#',
    };
    const { getByTestId } = render(
      <EmbeddedFeature1
        embeddedAsset={embeddedAsset}
        style={{}}
        title="title test"
        className="foo"
      />,
    );
    const element = getByTestId(TYPOGRAPHY_DEFAULT_TEST_ID);
    expect(element.classList).toContain(TypographyTypes.Headline4);
  });
  test('should render a Typography component when eyebrow is added', () => {
    const { getByTestId } = render(
      <EmbeddedFeature1 style={{}} eyebrow="eyebrow test" className="foo" />,
    );
    const element = getByTestId(TYPOGRAPHY_DEFAULT_TEST_ID);
    expect(element.classList).toContain(TypographyTypes.Headline6);
  });
  test('should render a Typography component when subtitle is added', () => {
    const { getByTestId } = render(
      <EmbeddedFeature1 style={{}} subtitle="subtitle test" className="foo" />,
    );
    const element = getByTestId(TYPOGRAPHY_DEFAULT_TEST_ID);
    expect(element.classList).toContain(TypographyTypes.Body);
  });
});
