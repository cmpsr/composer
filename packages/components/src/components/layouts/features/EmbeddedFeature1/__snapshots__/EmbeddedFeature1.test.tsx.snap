// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbeddedFeature1 should match snapshot 1`] = `
<div
  className="w-full h-full max-w-content mx-auto items-center"
  data-testid="box"
>
  <div
    className="text-left"
  >
    <h2
      className="typo-headline-4 mt-4"
      data-testid="typography"
    >
      title test
    </h2>
    <div
      className="flex justify-start mt-4"
    />
  </div>
</div>
`;
